### PR TITLE
[ADF-1586] Added docs for services

### PR DESCRIPTION
**Pull request by therealandeeee**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [x] Documentation
> - [ ] Other... Please describe:

**What is the new behaviour?**

Added docs for some more services and associated models.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

### CODE REVIEW

File: docs/bpm-user.model.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Details
+
+Instances of this class are returned by the methods of the
+[Bpm User service](bpm-user.service.md). It implements the
+`UserRepresentation` interface, which is defined in the
+[Alfresco JS API]h(ttps://github.com/Alfresco/alfresco-js-api/blob/master/src/alfresco-activiti-rest-api/docs/UserRepresentation.md).
+
+```ts
+class BpmUserModel implements UserRepresentation {
```
Comment (eromano) [inline docs/bpm-user.model.md]: I prefer to not have a copy of the model in the doc could be easily go out of sync

File: docs/ecm-user.model.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Details
+
+Instances of this class are returned by the methods of the
+[Ecm User service](ecm-user.service.md). It implements the `Person`
+interface, which is defined in the
+[Alfresco JS API](https://github.com/Alfresco/alfresco-js-api/blob/master/src/alfresco-core-rest-api/docs/Person.md).
+
+```ts
+class EcmUserModel implements Person {
```
Comment (eromano) [inline docs/ecm-user.model.md]: I prefer to not have a copy of the model in the doc could be easily go out of sync

File: docs/user-process.model.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Represents a Process Services user.
+
+## Details
+
+This class contains basic information about a Process Services user and
+is used throughout ADF to identify and list users (eg, to assign them to
+a task or to list them in search results).
+
+```ts
+class UserProcessModel implements LightUserRepresentation {
```
Comment (eromano) [inline docs/user-process.model.md]: I prefer to not have a copy of the model in the doc could be easily go out of sync

**Comment by therealandeeee** [comment]
I've kept the model doc files but replaced the class listings with links to the source file. I figured that it was useful to keep the Markdown files for the models because of the See Also bits and the links to the JS API but let me know if you would prefer to remove the model docs altogether.

**Pull request merged**